extract-wos-article.py

```diff
@@ -23,7 +30,54 @@
     "Current Publisher": "publisher",
 }
 
-YEAR_REGEX = re.compile(".*([0-9]{4})")
+REGEX = {
+    "year": re.compile(".*([0-9]{4})"),
+    "wos_id_number": re.compile("([0-9]{15})"),
+}
+CITING_SUMMARY_URL_PREFIX = "https://www.webofscience.com/wos/woscc/citing-summary/"
+
+
+async def extract_citing_summary(page: Page, article_id: str):
+    url = CITING_SUMMARY_URL_PREFIX + article_id
+    await page.goto(url, wait_until="networkidle")
+
+    curr_page = await page.query_selector(CITING_SELECTORS["current_page"])
+    curr_page_text = await curr_page.input_value()
+    curr_page_num = int(curr_page_text.strip())
+
+    final_page = await page.query_selector(CITING_SELECTORS["final_page"])
+    final_page_text = await final_page.inner_html()
+    final_page_num = int(final_page_text.strip())
+
+    articles = []
+    while True:
+        curr_articles_num = -1
+        curr_articles = []
+
+        while len(curr_articles) > curr_articles_num:
+            curr_articles_num = len(curr_articles)
+            await page.keyboard.press("PageDown")
+            curr_articles = await page.query_selector_all(
+                CITING_SELECTORS["citations_articles"]
+            )
+            time.sleep(0.1)
+
+        articles.extend(curr_articles)
+
+        if curr_page_num < final_page_num:
+            await page.click('[aria-label="Bottom Next Page"]')
+            time.sleep(2)
```

Comment:
@mrshu i'm not a big fan of this hack (there should be way such as 'wait till the page is fully loaded'), but it works. any improvement idea would be appreciated.

Reply:
Yeah, taking a look at the docs directly (https://playwright.dev/python/docs/api/class-page#page-click), one would expect the `page.click` to pick the right signal up but I guess that's not happening. 

Perhaps something could be done with [`page.locator`](https://playwright.dev/python/docs/api/class-locator#locator-wait-for)? If you won't be able to figure it out rather quickly (say in 30 minutes), I'd just leave it like it is -- it's good enough for now :slightly_smiling_face:

Reply:
@mrshu i've burnt some time with this without figuring out how to do it. i created the issue #8 TBD later.

Reply:
@JMatej Understood, feel free to leave it for now.